src/main/java/org/jabref/gui/maintable/StringTableColumn.java

```diff
@@ -0,0 +1,158 @@
+
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.value.ObservableValue;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+
+import org.jabref.gui.JabRefIcon;
+import org.jabref.logic.layout.LayoutFormatter;
+import org.jabref.logic.layout.format.LatexToUnicodeFormatter;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FieldName;
+import org.jabref.model.entry.FieldProperty;
+import org.jabref.model.entry.InternalBibtexFields;
+
+public class StringTableColumn extends MainTableColumn<String> {
+
+    private final List<String> bibtexFields;
+
+    private final boolean isIconColumn;
+
+    private final Optional<JabRefIcon> iconLabel;
+
+    private final Optional<BibDatabase> database;
+
+    private final LayoutFormatter toUnicode = new LatexToUnicodeFormatter();
+
+    public StringTableColumn(String columnName) {
+        super(columnName);
+        this.bibtexFields = Collections.emptyList();
+        this.isIconColumn = false;
+        this.iconLabel = Optional.empty();
+        this.database = Optional.empty();
+    }
+
+    public StringTableColumn(String columnName, List<String> bibtexFields, BibDatabase database) {
+        super(columnName);
+        this.bibtexFields = Collections.unmodifiableList(bibtexFields);
+        this.isIconColumn = false;
+        this.iconLabel = Optional.empty();
+        this.database = Optional.of(database);
+    }
+
+    public StringTableColumn(String columnName, List<String> bibtexFields, JabRefIcon iconLabel) {
+        super(columnName);
+        this.bibtexFields = Collections.unmodifiableList(bibtexFields);
+        this.isIconColumn = true;
+        this.iconLabel = Optional.of(iconLabel);
+        this.database = Optional.empty();
+    }
+
+    /**
+     * Get the table column name to be displayed in the UI
+     *
+     * @return name to be displayed. null if field is empty.
+     */
+    public String getDisplayName() {
+        if (bibtexFields.isEmpty()) {
+            return null;
+        }
+
+        StringJoiner joiner = new StringJoiner(FieldName.FIELD_SEPARATOR);
+        for (String field : bibtexFields) {
+            joiner.add(field);
+        }
+        return joiner.toString();
+    }
+
+    public List<String> getBibtexFields() {
+        return bibtexFields;
+    }
+
+    public boolean isIconColumn() {
+        return isIconColumn;
+    }
+
+    public boolean isFileFilter() {
+        return false; // Overridden in SpecialMainTableColumns for file filter columns
+    }
+
+    @Override
+    public ObservableValue<String> getColumnValue(BibEntryTableViewModel entry) {
+        if (bibtexFields.isEmpty()) {
+            return null;
+        }
+        boolean isNameColumn = false;
+
+        Optional<String> content = Optional.empty();
+        for (String field : bibtexFields) {
+            content = entry.getResolvedFieldOrAlias(field, database.orElse(null));
```

Comment:
Maybe stream with firstOrNull? or anyMatch?

Reply:
not possible because `isNameColumn` has to be set.